docs/source/deepinv.loss.rst

```diff
@@ -93,19 +93,58 @@ Training is implemented using :class:`deepinv.training.AdversarialTrainer` which
 
 Metrics
 --------
-Metrics are generally used to evaluate the performance of a model. Some of them can be used as training losses as well.
+Metrics are generally used to evaluate the performance of a model.
+
+Metrics inherit from the :class:`deepinv.loss.metric.Metric` baseclass, and take either ``x_net, x``
+for a full-reference metric or ``x_net`` for a no-reference metric. 
+
+.. note::
+
+    Metrics may also optionally take in measurements ``y``, the physics and the model.
+    The arguments are the same as in :class:`deepinv.loss.Loss`.
+
+
+All metrics can perform a standard set of pre and post processing, including
+operating on complex numbers, normalisation and reduction. See :class:`deepinv.loss.metric.Metric` for more details.
+
+.. note::
+
+    By default, metrics do not reduce over the batch dimension, as the usual usage is to average the metrics over a dataset yourself.
+    However, you can use the ``reduction`` argument to perform reduction, e.g. if the metric is to be used as a training loss.
+
+All metrics can be used as training losses as well by setting ``train_loss=True``.
+For example, ``MSE(train_loss=True)`` replaces :class:`torch.nn.MSELoss` and ``MAE(train_loss=True)`` replaces :class:`torch.nn.L1Loss`.
+
+Metrics can either be used directly or as the backbone for loss functions:
+
+.. doctest::
+
+    >>> import torch
+    >>> import deepinv as dinv
+    >>> m = dinv.metric.SSIM()
+    >>> x = torch.ones(2, 3, 16, 16) # B,C,H,W
+    >>> x_hat = x + 0.01
+    >>> m(x_hat, x) # Calculate metric for each image in batch
+    tensor([1.0000, 1.0000])
+    >>> m = dinv.metric.SSIM(reduction="sum")
+    >>> m(x_hat, x) # Sum over batch
+    tensor(1.9999)
+    >>> l = dinv.loss.MCLoss(metric=dinv.metric.SSIM(train_loss=True, reduction="mean")) # Use SSIM for training
 
 .. autosummary::
    :toctree: stubs
    :template: myclass_template.rst
    :nosignatures:
 
-        deepinv.loss.MSE
-        deepinv.loss.NMSE
-        deepinv.loss.PSNR
-        deepinv.loss.SSIM
-        deepinv.loss.LPIPS
-        deepinv.loss.NIQE
+        deepinv.loss.metric.Metric
```

Comment:
not sure this renaming is that useful - what is the idea behind?

Reply:
I worry that it is not backwards compatible

Reply:
see above